### PR TITLE
Contribution Guidelines Markdown

### LINKED ISSUE #475: A well crafted Contribution Guidelines
__NOTE__: DO NOT TAKE THIS ASSUMING IT TO BE A BEGINNER ISSUE THAT CAN BE DONE IN A COUPLE OF MINUTES

# REQUIREMENTS.

1. **Section A**: File Structure of the project

- Explain neatly

2.  **Section C**: How to get the Project?

- Install git
- Setup GitHub Account
- Fork 
- Clone
- Move inside the directory
- Set Upstream 
- Pull all the branches from upstream

3. **Section B**: How to set up the project?

-  Basic react stuff ( If anyone copies this line, I am going to mark the PR as spam😅)

4.  **Section C**: How to make changes and contribute to the project?

- Pull changes from Upstream
- Comment on an issue/Raise an issue
- Get it approved by the project maintainers
- Create a new feature branch (DO NOT name it MAIN or MASTER or anything with like `8904thfbn`)
- Do the changes
- Check the outcome
- Make a small clip or take screenshots
- Stage
- Commit
- Push to GitHub (Not to MAIN/MASTER)
- Make a PR

4. **Section D: PR Guidelines**:
- Add Appropriate Title
- Add Appropriate Description
- Screenshots/Clips

** Give an example of the above
- Rebase and Squash Commit if changes are suggested
- Do not comment back on the issue " Please check my PR. Maintainers will have a look as per their convenience"

# General Instructions:

- The above points are for reference. **Do not copy paste exactly!**

- Explain each of the commands with an appropriate description and a follow-up gif (perhaps showing the action) or an illustration

- Use codeblock for the commands

- Emojis are appreciated 

- All the gifs/screenshots/illustrations should be stored in `readme_assets/` with appropriate names

- *Only use Markdown. Refrain from using HTML tags inside Markdown*

### PR DESCRIPTION
**Pull request by kanishkaa24**
Closes #475 
Added all the sections supported with emojis and images. 
Added all the images in readme/assets.
Participant at DWOC, SWOC, CrossWoC.

### CODE REVIEW

**Comment by smaranjitghose** [comment]
A small change is required.

```
git add .
```
You missed the . for specifying all the files in the current path


Also, please try do a squash commit for this PR as well

File: ContributionGuidelines.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# [DOC2PEN](https://doc2pen.smaranjitghose.codes/)📝
+#### Doc2pen is an Open Source Software that converts your Typed Docs into Handwritten Docs.
+
+# Contribution Guidelines [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/dwyl/esta/issues) 🤝🏽🍀:
+
+## File Structure of the Project 🗃️📂
+
+### [Public Directory Structure](https://github.com/smaranjitghose/doc2pen/tree/master/public) 🧮✨
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:8]: Public Directory of not of any interest. We are just concerned with src

File: ContributionGuidelines.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+# [DOC2PEN](https://doc2pen.smaranjitghose.codes/)📝
+#### Doc2pen is an Open Source Software that converts your Typed Docs into Handwritten Docs.
+
+# Contribution Guidelines [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/dwyl/esta/issues) 🤝🏽🍀:
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:4]: Make the title please

File: ContributionGuidelines.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+- _redirects
+- robots.txt
+
+
+### [Src Directory Structure](https://github.com/smaranjitghose/doc2pen/tree/master/src) 🧮✨
+This folder contains all the main source code for the React application.
+
+Files/Folders that are directly located inside the Src Folder: 📥
+- app.js 
+- app.test.js
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:28]: It is better to explain this with a tree structure:

|-src
     |-... 
     |- 
        _- 

Now explain each one

File: ContributionGuidelines.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+### [Readme_Assets Directory Structure](https://github.com/smaranjitghose/doc2pen/tree/master/readme_assets)🧮✨
+This folder contains all the images used in Readme Markdown.
+
+## To Get the Project on your local machine 💻🧑‍💻👩‍💻
+- Install [Git](https://git-scm.com/downloads) 📥
+- Setup [Github](https://github.com/) Account 📇
+- Fork [this](https://github.com/smaranjitghose/doc2pen) Project 🍴
+<p align = "center">
+  <img src="https://github.com/kanishkaa24/doc2pen/blob/master/readme_assets/Fork%20Project.png?raw=true" width="500" height="200"/>
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:56]: Give relative path for assets please

File: ContributionGuidelines.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+- Add a reference(remote) to the original repository.
+
+   ```
+   git remote add upstream https://github.com/smaranjitghose/doc2pen.git
+   ```
+
+- Always take a pull from the upstream repository to your master branch to keep it at par with the main project(updated repository).
+
+   ```
+   git pull upstream master
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:80]: This is explained later. So the upstream part can be saved for the contributing part

File: ContributionGuidelines.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+Files/Folders that are directly located inside the Src Folder: 📥
+- .DS_Store
+- images
+- js
+
+### [Readme_Assets Directory Structure](https://github.com/smaranjitghose/doc2pen/tree/master/readme_assets)🧮✨
+This folder contains all the images used in Readme Markdown.
+
+## To Get the Project on your local machine 💻🧑‍💻👩‍💻
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:51]: `How to get the project on your local machine?`

Keep in mind the capitalization

**Comment by kanishkaa24** [comment]
> A small change is required.
> 
> ```
> git add .
> ```
> 
> You missed the . for specifying all the files in the current path
> 
> Also, please try do a squash commit for this PR as well

Sure Sir

File: ContributionGuidelines.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+   ```
+   git pull upstream master
+   ```
+
+<p align="center">
+  <img src="https://github.com/kanishkaa24/doc2pen/blob/master/readme_assets/Cloning%20Project%20(1).png?raw=true"/>
+</p>
+
+## To Run the Project on your Local Machine ⌨️🖥️🖱️
+- Navigate to the project.
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:88]: Already covered above.

`cd Doc2Pen`

File: ContributionGuidelines.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+- Run `npm install`.
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:89]: Write the relevance of the command please😅 followed by the command itself within a code block

File: ContributionGuidelines.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+- Run `npm start`.
+- Navigate to http://localhost:3000 to view it in the browser.
+
+## To Make Changes in the Project 🎨👩‍🎨👨‍🎨
+- Pull changes from Upstream.
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:94]: Add Upstream can be dicussed here

File: ContributionGuidelines.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+- Get it approved and assigned by the project maintainers.
+- Create a new feature branch (DO NOT name it MAIN or MASTER or anything random).
+   ```
+   git checkout -b <your_branch_name>
+   ```
+- Do the changes.
+- Check the outcome.
+- Make a small clip or take screenshots.
+<p align="center">
+  <img src="https://github.com/smaranjitghose/doc2pen/blob/master/readme_assets/Contribute.png?raw=true" height="200" width="200"/>
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:108]: This illustration is not valid for this context. Perhaps put it on the top

File: ContributionGuidelines.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+   ```
+   git add
+   ```
+- Commit your changes.
+   ```
+   git commit -m "Relevant message"
+   ```
+- Push the changes.
+   ```
+   git push -u origin <your_branch_name>
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:121]: avoid the -u flag

File: ContributionGuidelines.md [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+   ```
+- Make a Pull Request.
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:123]: Write it in a single line, how to create a PR

**Comment by kanishkaa24** [comment]
I am not able to squash the commits. Is the current one fine?

File: ContributionGuidelines.md [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+    To create a pull request, click on `compare and pull requests`.
+ <p align="center">
+  <img src="https://github.com/smaranjitghose/doc2pen/blob/master/readme_assets/ComparePR.png?raw=true"/>
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:126]: Use relative path please

File: ContributionGuidelines.md [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+</p>
+
+## To Make a Pull Request, Follow the below guidelines ✅
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:129]: Number the sections, please.

File: ContributionGuidelines.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+- Add an appropriate title.
+- Add an appropriate description of your work.
+- Add images/screenshots.
+- Mention the issue.
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:133]: Write a line on how to do it

File: ContributionGuidelines.md [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+<p align="center">
+  <img src="https://github.com/kanishkaa24/doc2pen/blob/master/readme_assets/PR.JPG?raw=true" height="350" width="450"/>
+</p>
+
+### DON'Ts ❎
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:138]: Make the heading as NOTE

File: ContributionGuidelines.md [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+- Rebase and Squash Commit if changes are suggested.
```
Comment (smaranjitghose) [inline ContributionGuidelines.md:139]: Write the commands for this

**Review by smaranjitghose — CHANGES_REQUESTED** [review]

**Comment by smaranjitghose** [comment]
> I am not able to squash the commits. Is the current one fine?

What is the error?

File: ContributionGuidelines.md (hunk 1 above)
Comment (kanishkaa24) [reply ContributionGuidelines.md:8]: So Shall I remove public directory?

File: ContributionGuidelines.md (hunk 2 above)
Comment (kanishkaa24) [reply ContributionGuidelines.md:4]: I don't understand

File: ContributionGuidelines.md (hunk 3 above)
Comment (kanishkaa24) [reply ContributionGuidelines.md:28]: Sure

File: ContributionGuidelines.md (hunk 4 above)
Comment (kanishkaa24) [reply ContributionGuidelines.md:56]: Sure

File: ContributionGuidelines.md (hunk 5 above)
Comment (kanishkaa24) [reply ContributionGuidelines.md:80]: Sure

**Comment by kanishkaa24** [comment]
> > I am not able to squash the commits. Is the current one fine?
> 
> What is the error?

I edited the markdown directly on github and not on my local machine.

File: ContributionGuidelines.md (hunk 2 above)
Comment (smaranjitghose) [reply ContributionGuidelines.md:4]: Keep the first line of the file as #Contribution Guidelines

**Comment by smaranjitghose** [comment]
> > > I am not able to squash the commits. Is the current one fine?
> > 
> > 
> > What is the error?
> 
> I edited the markdown directly on github and not on my local machine.

Either way, commits are there😅. Even github is a server for git. Just pull the changes locally, do a rebase, make a squash commit and update the pr

**Comment by kanishkaa24** [comment]
> > > > I am not able to squash the commits. Is the current one fine?
> > > 
> > > 
> > > What is the error?
> > 
> > 
> > I edited the markdown directly on github and not on my local machine.
> 
> Either way, commits are there😅. Even github is a server for git. Just pull the changes locally, do a rebase, make a squash commit and update the pr

Can I make a new PR for this and close this one?

**Comment by ashuvssut** [comment]
@kanishkaa24 I would suggest to work on your local machine then you can squash commit easily.
(I dont know if you can squash commit on GitHub or not😅)

Steps to do a squash commit
```git
# i can see that there are total 18 commits
# first go back (move HEAD) to your eighteenth last commit
git reset --soft HEAD~18
# now head is behind the eighteenth commit 7d0f47d

# now perform "commit" to squash commit
git commit -m "new commit message"

#force push you new commit deleting  these two commits fa0096d and 67c07e3 on your forked remote repo
git push -f

```

**Comment by ashuvssut** [comment]
@smaranjitghose When I was working on the projectStructure issue, there, I provided a [Project_Structure.md](https://github.com/smaranjitghose/doc2pen/blob/master/Project_Structure.md) file. It might need a small update but it contains more and precise info about the project structure. I think providing a link from contributing.md to [Project_Structure.md](https://github.com/smaranjitghose/doc2pen/blob/master/Project_Structure.md) file would be nice

**Comment by kanishkaa24** [comment]
> @kanishkaa24 I would suggest to work on your local machine then you can squash commit easily.
> (I dont know if you can squash commit on GitHub or not😅)
> 
> Steps to do a squash commit
> 
> ```shell
> # i can see that there are total 18 commits
> # first go back (move HEAD) to your eighteenth last commit
> git reset --soft HEAD~18
> 
> # now head is behind the eighteenth commit 7d0f47d
> 
> # now perform "commit" to squash commit
> git commit -m "new commit message"
> 
> #force push you new commit deleting  these two commits fa0096d and 67c07e3 on your forked remote repo
> git push -f
> ```
> 
> > Don't make changes to you master branch. keep master branch to get updates from the `upstream` repo. you can make an new branch from `master` and then make changes->`commit`->`push` that branch to you forked repo. then simply create a PR from that new branch

Thanks

**Comment by kanishkaa24** [comment]
@smaranjitghose 
I am thinking about deleting my earlier fork and forking it again. Can I proceed that way? I'll finish with all changes ASAP.

**Comment by smaranjitghose** [comment]
> @smaranjitghose
> I am thinking about deleting my earlier fork and forking it again. Can I proceed that way? I'll finish with all changes ASAP.

sure

**Pull request closed without merging**